tests/neg-custom-args/fatal-warnings/i9241.scala

```diff
@@ -16,6 +16,7 @@ final class Baz private (val x: Int) extends AnyVal {
   def unary_- : Baz = ???
   def unary_+[T] : Baz = ???
   def unary_!() : Baz = ??? // error
+  def `unary_!_=`() : Baz = ??? // ok
```

Comment:
This file tests that the definitions of unary operators are well defined. If not an error is emitted.  

``def `unary_!_=`()`` is clearly not well defined as it does not receive the value of the RHS.

Reply:
Here is a full version of the equivalent test case for `unarry_X_=`

<details>
<summary> Test case

`tests/neg-custom-args/fatal-warnings/i9241b.scala`
</summary>

```scala
class Foo {
  def `unary_!_=`() : Foo = this // error
  def `unary_+_=`(using Int)(): Foo = this // error
  def `unary_-_=`()(implicit i: Int): Foo = this // error
  def `unary_~_=`[T](): Foo = this // error
}

class Bar {
  def `unary_!_=`(value: Int) : Baz = ???
  def `unary_-_=`[T](value: T)(using T): Baz = ???
  def `unary_~_=`() : Baz = ??? // error
  def `unary_+_=`(value: Int, value2: Int) : Baz = ??? // error
}

final class Baz private (val x: Int) extends AnyVal {
  def `unary_!_=`() : Baz = ??? // error
  def `unary_+_=`(value: Int) : Baz = ???
}

extension (x: Int)
  def `unary_!_=` : Int = ??? // error
  def `unary_+_=`[T] : Int = ??? // error
  def `unary_-_=`() : Int = ??? // error
  def `unary_~_=`(using Int) : Int = ??? // error
end extension

extension (x: Long)
  def `unary_!_=`(value: Int): Int = ???
  def `unary_+_=`[T](value: T) : Int = ???
  def `unary_-_=`(value: Int) : Int = ???
  def `unary_~_=`(value: Int)(using Int) : Int = ???
end extension

extension [T](x: Byte)
  def `unary_!_=` : Int = ??? // error
  def `unary_+_=`[U] : Int = ??? // error
  def `unary_-_=`() : Int = ??? // error
  def `unary_~_=`(using Int) : Int = ??? // error
end extension

extension [T](x: Short)
  def `unary_!_=`(value: Int): Int = ???
  def `unary_+_=`[U](value: T) : Int = ???
  def `unary_-_=`(value: Int) : Int = ???
  def `unary_~_=`(value: Int)(using Int) : Int = ???
end extension

extension (using Int)(x: Float)
  def `unary_!_=` : Int = ??? // error
  def `unary_+_=`[T] : Int = ??? // error
  def `unary_-_=`() : Int = ??? // error
  def `unary_~_=`(using Int) : Int = ??? // error
end extension

extension (using Int)(x: Double)
  def `unary_!_=`(value: Int): Int = ???
  def `unary_+_=`[T](value: T) : Int = ???
  def `unary_-_=`(value: Int) : Int = ???
  def `unary_~_=`(value: Int)(using Int) : Int = ???
end extension
```

</details>

Reply:
See `RefChecks.checkUnaryMethods`

Reply:
I just wanted to make sure and show that we can now parse this. We can break it out in a different test, if you think that's better.

Reply:
Then this is not the tests file where this should go. It will be extremely misleading to the next person that looks at the test. I suggest we put its own test file and include a comment that says that we can define it but we will not be able to use it.

Reply:
We can enhance the i9241 checks in a later PR.

Reply:
I moved the test and will add the checks in a subsequent PR.